### PR TITLE
fix: not pledged

### PR DESCRIPTION
**Pull request by ashitakah**
# 🤖 Linear

Closes GRT-8

The problem is that according to the report when the dispute is in non-resolution, everyone can access a pledge. The issue is that it doesn't check if that user made a previous pledge and anyone could subtract it in that state. 

I have opted to create a mapping that both this in no resolution or if when you make a pledge you put it in true and if not in false, I do not know if it solves the problem 100%. 

REVIEW!

### CODE REVIEW

File: solidity/contracts/extensions/BondEscalationAccounting.sol [hunk 1]
```diff
@@ -33,6 +35,8 @@ contract BondEscalationAccounting is AccountingExtension, IBondEscalationAccount
 
     pledges[_disputeId][_token] += _amount;
 
+    pledgerHasPledged[_disputeId][_pledger] = true;
```
Comment (0xShaito) [inline solidity/contracts/extensions/BondEscalationAccounting.sol:38]: Can't the same pledger pledge multiple times? This would kill that right?

**Review by 0xShaito — CHANGES_REQUESTED** [review]

File: solidity/contracts/extensions/BondEscalationAccounting.sol (hunk 1 above)
Comment (ashitakah) [reply solidity/contracts/extensions/BondEscalationAccounting.sol:38]: Yep, but as far I know if the resolution is non, the user can claim their pledge 1 time, so with that will be solve I guess.

Comment (0xShaito) [reply solidity/contracts/extensions/BondEscalationAccounting.sol:38]: No because they will only be able to claim 1 pledge. Losing the rest of the money

Comment (0xShaito) [reply solidity/contracts/extensions/BondEscalationAccounting.sol:38]: Missed the check for amount of pledges. You are right! Tests are failing though

File: solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol [hunk 2]
```diff
@@ -462,6 +476,8 @@ contract BondEscalationAccounting_Unit_ClaimEscalationReward is BaseTest {
 
     bondEscalationAccounting.forTest_setPledge(_disputeId, token, _amount * _pledges);
 
+    bondEscalationAccounting.forTest_setPledgerHasPledged(_disputeId, pledger, true);
```
Comment (0xShaito) [inline solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol:479]: Do we have a test for someone who pledged multiple times?

File: solidity/test/integration/BondEscalation.t.sol [hunk 3]
```diff
@@ -356,8 +356,8 @@ contract Integration_BondEscalation is IntegrationBase {
 
     // Step 10: Participants claim their rewards
     // Test: The requester has paid out the reward and is left with no balance
-    _bondEscalationAccounting.claimEscalationReward(_disputeId, requester);
-    assertEq(_bondEscalationAccounting.balanceOf(requester, usdc), 0, 'Mismatch: Requester balance');
+    // _bondEscalationAccounting.claimEscalationReward(_disputeId, requester);
```
Comment (0xShaito) [inline solidity/test/integration/BondEscalation.t.sol:359]: Why comment?

File: solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol [hunk 4]
```diff
@@ -533,4 +533,59 @@ contract BondEscalationAccounting_Unit_ClaimEscalationReward is BaseTest {
...
+    vm.assume(_amount > 0);
+    vm.assume(_pledges > 0 && _pledges < 30);
+
+    _amount = bound(_amount, 0, type(uint128).max / _pledges);
+
+    bondEscalationAccounting.forTest_setEscalationResult(
+      _disputeId, _requestId, token, _amount, IBondEscalationModule(_bondEscalationModule)
+    );
+
+    bondEscalationAccounting.forTest_setPledge(_disputeId, token, _amount * _pledges * 2);
```
Comment (0xShaito) [inline solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol:553]: Can we have a different number of pledges for and against so we are sure we are adding both numbers correctly?

**Review by 0xShaito — CHANGES_REQUESTED** [review]

File: solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol (hunk 4 above)
Comment (ashitakah) [reply solidity/test/unit/modules/dispute/BondEscalationAccounting.t.sol:553]: 👍

File: solidity/test/integration/BondEscalation.t.sol (hunk 3 above)
Comment (ashitakah) [reply solidity/test/integration/BondEscalation.t.sol:359]: mistake

**Review by 0xJabberwock — APPROVED** [review]

**Review by 0xShaito — APPROVED** [review]

**Pull request merged**